app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,29 @@
+class PostsController < ApplicationController
+  require "will_paginate/array"
+  PAGE_SIZE = 4
+
+  def index
+    page = (params[:page] || '1').to_i
+
+    client = Tumblr::Client.new({
```

Comment:
maybe this could be extracted into an adapter wrapping the client setup and results fetching? I'm thinking a `Blog` class under `app/adapters/`. It could return the posts as a struct, and here in `Posts#index` you could have a simple `@posts = Blog.posts`